### PR TITLE
Fixed config resolve for fragments

### LINKED ISSUE #365: Query execution not working on Fragments
When I go into my query definitions in Javascript and press ALT+Enter to invoke the "Edit GraphQL Fragment" action, the fragment opens but I cannot execute it. This used to work.

The .graphqlconfig that lies in the project root is not regarded. 

Bildschirmfoto von 2020-08-18 20-54-49
The dropdown is empty and I cannot execute the query.

Executing queries from scratch files keeps working.

Ubuntu 18.04.5 LTS
IntelliJ current 2020.2
JS GraphQL 2.5.0

 [Minimized reproducing example](https://github.com/jimkyndemeyer/js-graphql-intellij-plugin/files/5092270/gql-bug.zip)

### PR DESCRIPTION
**Pull request by vepanimas**
Fixes #365

### CODE REVIEW

File: src/main/com/intellij/lang/jsgraphql/GraphQLFileType.java [hunk 1]
```diff
@@ -20,54 +20,55 @@
...
+    public static boolean isGraphQLScratchFile(Project project, VirtualFile file) {
+        if (ScratchUtil.isScratch(file)) {
+            final PsiManager psiManager = PsiManager.getInstance(project);
+            try {
+                final PsiFile psiFile = psiManager.findFile(file);
+                if (psiFile != null && psiFile.getFileType() == GraphQLFileType.INSTANCE) {
+                    return true;
+                }
+            } catch (ProcessCanceledException e) {
+                // can be thrown from psiManager.findFile
```
Comment (anstarovoyt) [inline src/main/com/intellij/lang/jsgraphql/GraphQLFileType.java:68]: should be rethrown, I believe

File: src/main/com/intellij/lang/jsgraphql/psi/GraphQLPsiUtil.java [hunk 2]
```diff
@@ -48,4 +53,54 @@ public static String getTypeName(PsiElement psiElement, @Nullable Ref<GraphQLIde
         return null;
     }
 
+    @Nullable
+    public static VirtualFile getVirtualFileFromPsiFile(@Nullable PsiFile containingFile) {
```
Comment (anstarovoyt) [inline src/main/com/intellij/lang/jsgraphql/psi/GraphQLPsiUtil.java:57]: com.intellij.psi.util.PsiUtilCore#getVirtualFile ?

**Review by anstarovoyt — APPROVED** [review]

File: src/main/com/intellij/lang/jsgraphql/psi/GraphQLPsiUtil.java (hunk 2 above)
Comment (vepanimas) [reply src/main/com/intellij/lang/jsgraphql/psi/GraphQLPsiUtil.java:57]: It does
```
public static VirtualFile getVirtualFile(@Nullable PsiElement element) {
    // optimisation: call isValid() on file only to reduce walks up and down
    if (element == null) {
      return null;
    }
    if (element instanceof PsiFileSystemItem) {
      return element.isValid() ? ((PsiFileSystemItem)element).getVirtualFile() : null;
    }
```
and it differs from the existing behavior here.

File: src/main/com/intellij/lang/jsgraphql/GraphQLFileType.java (hunk 1 above)
Comment (vepanimas) [reply src/main/com/intellij/lang/jsgraphql/GraphQLFileType.java:68]: It shouldn't be caught at all here, but I believe there is a reason why it's here. I'll check it.

**Pull request merged**